v21.2/deploy-lambda-function.md

```diff
@@ -0,0 +1,260 @@
+
+## Step 1. Create a {{ site.data.products.serverless }} cluster
+
+{% include cockroachcloud/quickstart/create-a-free-cluster.md %}
+
+## Step 2. Set up your cluster connection
+
+1. Once your cluster is created, the **Connection info** dialog displays. Follow the instructions to do the following:
+
+    1. Download the CockroackDB client.
+
+    1. Download the CA certificate.
+
+    1. Connect to the database with the [built-in SQL shell](cockroach-sql.html).
+
+1. In the SQL shell, create the `bank` database that your Lambda function will use:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ sql
+    > CREATE DATABASE bank;
+    ~~~
+
+1. Exit the SQL shell:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ sql
+    > \q
+    ~~~
+
+## Step 3. Get the sample code
+
+1. Clone the sample code's GitHub repo:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ git clone https://github.com/cockroachlabs/examples-aws-lambda/
+    ~~~
+
+1. In the `examples-aws-lambda/python` directory, create a new folder called `certs`.
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ cd examples-aws-lambda/python
+    ~~~
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ mkdir certs
+    ~~~
+
+1. Copy the certificate that you downloaded for your cluster to the `certs` directory:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ cp ~/.postgresql/root.crt ./certs/root.crt
+    ~~~
+
+    The project directory structure should look like this:
+
+    ~~~
+    ├── Dockerfile
+    ├── certs
+    │   └── root.crt
+    ├── init_db.py
+    └── requirements.txt
+    ~~~
+
+## Step 4. Build the Docker image
+
+1. Build the Docker image locally:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ docker build -t init-crdb .
+    ~~~
+
+    This command builds a Docker image based on the [AWS Lambda base image for Python](https://gallery.ecr.aws/lambda/python).
+
+1. (Optional) Run the Docker image locally:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    docker run \
+    -e DATABASE_URL="postgresql://<username>:<password>@<serverless-host>:26257/bank?sslmode=verify-full&sslrootcert=certs/root.crt&options=--cluster%3D<cluster-name>" \
+    -p 9000:8080 init-crdb
+    ~~~
+
+    Note that this command sets the `DATABASE_URL` environment variable to the cluster connection string, where:
+    - `<username>` is the SQL user. By default, this is your {{ site.data.products.db }} account username.
+    - `<password>` is the password for the SQL user. The password will be shown only once in the **Connection info** dialog after creating the cluster.
+    - `<serverless-host>` is the hostname of the Serverless cluster.
+    - `<cluster-name>` is the short name of your cluster plus the tenant ID. For example, `funny-skunk-3`. The `<cluster-name>` is used to identify your tenant cluster on a multitenant host.
```

Comment:
I think we shouldn't mention the "tenant ID" in our docs. It's an implementation detail that's subject to change, and it introduces an extra concept that users shouldn't need to understand. Instead, users should be able to grab this info right from the connection parameters dialog.

Reply:
See also https://cockroachlabs.atlassian.net/browse/DOC-1856

Reply:
We added it in because a lot of users saw `<cluster-name>` and thought that just meant the short name. There was a long discussion [here](https://cockroachlabs.atlassian.net/browse/CC-5285) about it. I'm fine with removing tenant ID from docs, just noting that this distinction is definitely tripping people up.